infoschema/tables.go

```diff
@@ -1949,6 +1955,10 @@ func DataForAnalyzeStatus() (rows [][]types.Datum) {
 		} else {
 			startTime = types.Time{Time: types.FromGoTime(job.StartTime), Type: mysql.TypeDatetime}
 		}
+		if checker != nil && !checker.RequestVerification(ctx.GetSessionVars().ActiveRoles, job.DBName, job.TableName, "", mysql.AllPrivMask) {
+			job.Unlock()
```

Comment:
Why not move this block before assigning `startTime`?

Reply:
I have been insert `rows = append...` into `if checker...`, like this 
```
if checker == nil || checker.RequestVerification(ctx.GetSessionVars().ActiveRoles, job.DBName, job.TableName, "", mysql.AllPrivMask) {
rows = append............
}
```
it seems more clean. after that, it can't move this block before assigning startTime.